pkgs/tools/security/metasploit/default.nix

```diff
@@ -15,17 +15,22 @@ let
   };
 in stdenv.mkDerivation rec {
   pname = "metasploit-framework";
-  version = "6.3.43";
+  version = "6.3.44";
 
   src = fetchFromGitHub {
     owner = "rapid7";
     repo = "metasploit-framework";
-    rev = version;
-    sha256 = "sha256-IWdSxXhJFGbNTo9xqEjBJNEWB6imzdwzgnXoUbfao/g=";
+    rev = "refs/tags/${version}";
+    hash = "sha256-LBmyFE617G2ryEAp5fig3G2T/goa9NXUggu0XH4X1xs=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [ (python3.withPackages (ps: [ ps.requests ])) ];
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    (python3.withPackages (ps: [ ps.requests ]))
+  ];
```

Comment:
9 lines of unrelated refactoring.

Reply:
Is this an issue? I often get encouraged to do this kind of refactoring as part of a package update ( e.g. https://github.com/NixOS/nixpkgs/pull/266533#discussion_r1398411940 ).

Reply:
When I read a commit that bumps a version, I expect it to only do that. My mind is focused on "what's the required changes needed to do the update?". If there's unrelated changes, that's distracting. Also, version bumps pull in new code from the internet, which I think requires more careful review than reformatting changes.

Reformatting in a version bump PR is fine of course, but I think they should be a separate commit, where one can say what is done and why. (In this case, I don't even know why those 9 lines are changed. Is it an automated tool that did it? Or did the author do it manually because they think it improves readability?)

Does it make sense? Thoughts?

> I often get encouraged to do this kind of refactoring as part of a package update ( e.g. https://github.com/NixOS/nixpkgs/pull/266533#discussion_r1398411940 ).

Isn't that just about "sha256" -> "hash"? The line gets changed anyway in a version bump, so I think that's OK.

Reply:
> Isn't that just about "sha256" -> "hash"? The line gets changed anyway in a version bump, so I think that's OK.

Technically are changing `sha256` to `hash` and the SRI hash itself two changes. But that's usually combined like adding `refs/tags/` to avoid branch/tag issue when updating. The update bot is doing this anyways.

I think that it's a good trade-off to include formatting changes, high level changes and alike in updates as this will keep the amount of commits low and the maintenance work-flow straight forward and time-efficient. Of course, one could always squash the commit before merging (as we don't let GitHub do this) but would add additional tasks on the reviewer's plate.

IMHO, in regard of back-porting the argument of keeping the commits atomic weighted much more that changes which go to `master` only.

For sure, the line is a bit blurry what changes or amount of changes should go in one commit. Thus, it's good that a reviewer can always ask to split as the committer's opinion and the reviewers can differ.

> Is it an automated tool that did it?

Was done manually as `nixpkgs-fmt` will not go that far.

> Or did the author do it manually because they think it improves readability?

Readability? Perhaps, as always, that depends on the point of view. It was done to follow our code conventions a little closer and also to have easier to read diffs in the future. Short lists or attribute sets can be written on one line but this can't be alphabetically sorted by default tools.